### PR TITLE
Update PCT

### PR DESCRIPTION
**Pull request by timja**

### CODE REVIEW

**Comment by timja** [comment]
```
➜  plugin-compat-tester git:(162dffc) git bisect good
ae2dd7130a56f7298437cd0a72cd7d4a06eb3fe3 is the first bad commit
commit ae2dd7130a56f7298437cd0a72cd7d4a06eb3fe3
Author: imonteroperez <imontero@cloudbees.com>
Date:   Wed Sep 8 11:51:51 2021 +0200

    Bugfix on dealing with non-executed test phase due collateral issues

 .../org/jenkins/tools/test/PluginCompatTester.java | 13 ++++++----
 .../tools/test/util/ExecutedTestNamesDetails.java  | 29 ++++++++++++++++------
 .../tools/test/util/ExecutedTestNamesSolver.java   | 22 ++++++++--------
 .../jenkins/tools/test/PluginCompatTesterTest.java | 24 ++++++++++++++++++
 .../src/test/resources/bad-surefire-exclusion-list |  1 +
 5 files changed, 67 insertions(+), 22 deletions(-)
 create mode 100644 plugins-compat-tester/src/test/resources/bad-surefire-exclusion-list
```

**Comment by timja** [comment]
According to https://ci.jenkins.io/job/Tools/job/bom/job/PR-863/2/pipeline-console/?selected-node=332

The build is erroring when injected tests fail

**Comment by jetersen** [comment]
Perhaps the injected tests do not add value to PCT?

**Comment by timja** [comment]
> Perhaps the injected tests do not add value to PCT?

Yeah possibly, we already ignore them in bom https://github.com/jenkinsci/bom/blob/master/pct.sh#L41

I'm reverting the PR affecting us in our fork of PCT anyway https://github.com/jenkinsci/plugin-compat-tester/pull/343

Upgrading should be dealt with separately to this issue imo

**Comment by timja** [comment]
I manually deployed a SNAPSHOT because the folder is setup to merge with master bringing in changes that cause bom to fail.

I tried reverting the PR that caused issues but was having difficulty.

this could work in the meantime what do people think?

**Comment by jetersen** [comment]
Whatever works but could you get a link to a commit sha at least of what is in the snapshot? 🤔

Would be nice to finally land a proper PCT fix?

File: prep.sh [hunk 1]
```diff
@@ -41,9 +41,10 @@ do
 done
 
 # TODO find a way to encode this in some POM so that it can be managed by Dependabot
-version=0.5.1-rc1066.9cfd30140d30 # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/276
-pct=$HOME/.m2/repository/org/jenkins-ci/tests/plugins-compat-tester-cli/${version}/plugins-compat-tester-cli-${version}.jar
-[ -f $pct ] || $MVN dependency:get -Dartifact=org.jenkins-ci.tests:plugins-compat-tester-cli:${version}:jar -DremoteRepositories=https://repo.jenkins-ci.org/public/,https://repo.jenkins-ci.org/incrementals/ -Dtransitive=false
+version=0.5.1-SNAPSHOT # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/276
```
Comment (timja) [inline prep.sh:44]:
```suggestion
version=0.5.1-SNAPSHOT # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/344 (c9c684eab7e8de91650ad1c96ded6489c42910f1)
```

**Comment by timja** [comment]
done

**Review by jetersen — APPROVED** [review]

File: failFast [hunk 2]
```diff
@@ -1 +1 @@
-true
+false
```
Comment (jglick) [inline failFast:1]: revert when ready

File: prep.sh [hunk 3]
```diff
@@ -41,9 +41,10 @@ do
 done
 
 # TODO find a way to encode this in some POM so that it can be managed by Dependabot
-version=0.5.1-rc1066.9cfd30140d30 # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/276
-pct=$HOME/.m2/repository/org/jenkins-ci/tests/plugins-compat-tester-cli/${version}/plugins-compat-tester-cli-${version}.jar
-[ -f $pct ] || $MVN dependency:get -Dartifact=org.jenkins-ci.tests:plugins-compat-tester-cli:${version}:jar -DremoteRepositories=https://repo.jenkins-ci.org/public/,https://repo.jenkins-ci.org/incrementals/ -Dtransitive=false
+version=0.5.1-SNAPSHOT # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/344 (c9c684eab7e8de91650ad1c96ded6489c42910f1)
```
Comment (jglick) [inline prep.sh:44]: Better to get https://github.com/jenkinsci/plugin-compat-tester/pull/344 to pass tests—by ignoring stuff if necessary.

**Comment by jglick** [comment]
See #657. Ought to remain in draft until it is picking up an incremental build (if not an actual release).

File: prep.sh (hunk 3 above)
Comment (timja) [reply prep.sh:44]: ha, my last attempt of just reverting the PCT change that affected bom didn't go so well as there were other changes. Unless there's a simpler fix I'd probably just revert every PR in the last year =/. 

pr-merge pain -.-, can't build an isolated PR

Not sure if I'll have the time to fix properly.

Comment (jglick) [reply prep.sh:44]: Ack. PCT needs a lot of work.

File: failFast (hunk 2 above)
Comment (timja) [reply failFast:1]:
```suggestion
true
```

File: prep.sh [hunk 4]
```diff
@@ -41,10 +41,9 @@ do
 done
 
 # TODO find a way to encode this in some POM so that it can be managed by Dependabot
-version=0.5.1-SNAPSHOT # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/344 (c9c684eab7e8de91650ad1c96ded6489c42910f1)
-timestamp=0.5.1-20220130.152941-4
-pct=$HOME/.m2/repository/org/jenkins-ci/tests/plugins-compat-tester-cli/${version}/plugins-compat-tester-cli-${timestamp}.jar
-[ -f $pct ] || $MVN dependency:get -Dartifact=org.jenkins-ci.tests:plugins-compat-tester-cli:${timestamp}:jar -DremoteRepositories=https://repo.jenkins-ci.org/public/,https://repo.jenkins-ci.org/incrementals/ -Dtransitive=false
+version=1152.vafc19b26d5aa # TODO https://github.com/jenkinsci/plugin-compat-tester/pull/345
```
Comment (jglick) [inline prep.sh:44]: https://github.com/jenkinsci/plugin-compat-tester/pull/345 oof

**Pull request merged**